api/views/CloudRecordView.py

```diff
@@ -139,6 +132,29 @@ def _signer_is_valid(self, signer_dn):
                 # Continue looping through provider list, looking
                 # for a match in the remaining site JSON
 
+        # Return the hostnames we were able to extract.
+        return provider_hostnames
+
+    def _get_indigo_providers(self):
+        """Return a list of registered INDIGO Resource Provider hostnames."""
+        # Get the JSON from the CMDB.
+        provider_json = self._get_provider_json_indigo_cmdb()
+        # Return any hostnames we can parse from the provider JSON.
+        return self._parse_hostnames_indigo_cmdb(provider_json)
+
+    def _signer_is_valid(self, signer_dn):
+        """Return True if signer's host is listed as a Resource Provider."""
+        # Get the hostname from the DN
+        signer_split = signer_dn.split("=")
+        signer = signer_split[len(signer_split)-1]
+
+        if signer_dn in settings.BANNED_FROM_POST:
+            self.logger.info("Host %s is banned.", signer)
+            return False
+
+        if signer in self._get_indigo_providers():
+            return True
```

Comment:
Should this at least log to DEBUG?

Reply:
It probably should. I put it in as `info` to match the other returns in this method